examples/src/utils.js

```diff
@@ -0,0 +1,327 @@
+import { VASTClient, VASTTracker } from '@dailymotion/vast-client';
+const mainContent =
+  'https://dmplayer.storage.googleapis.com/tech_test/midnight_sun_720p.mp4';
+const videoPlayer = document.getElementById('myVideo');
+const variationImg = document.getElementById('variationImg');
+const variationContainer = document.getElementById('variationContainer');
+
+//vastTracker for linear ads
+let vastTracker = null;
+//vastTracker for companion ads
+let companionVastTracker = null;
+
+/**
+ * Get the first valid creative from an ad
+ */
+export const getCreative = (ad) => {
+  if (!ad.creatives) {
+    return null;
+  }
+  const validCreative = ad.creatives.find((creative) => creative.mediaFiles);
+  return validCreative || null;
+};
+
+/**
+ * Get the URL of the first media file in the creative
+ */
+
+export const getMediaFileUrl = (creative) => {
+  const mediaFiles = creative.mediaFiles;
+  if (!mediaFiles.length) {
+    return null;
+  }
+  const mediafile = mediaFiles.find((mediaFile) => mediaFile.fileURL);
+  return mediafile.fileURL || null;
+};
+
+/**
+ * @returns a promise that resolves with the parsed VAST
+ */
+
+export const parsePreroll = async () => {
+  const VAST = 'https://statics.dmcdn.net/h/html/vast/simple-inline.xml';
+
+  let vastClient = new VASTClient();
+  try {
+    return vastClient.get(VAST);
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+/**
+ * play a preroll
+ */
+
+export const playPreroll = async () => {
+  const parsedVAST = await parsePreroll();
+  // here we a using the first ad from the parsed vast that contain creatives,
+  // however, this logic can depend on you player implementation logic
+  const validAd = parsedVAST.ads.find((ads) => ads.creatives.length > 0);
+  if (!validAd) {
+    return;
+  }
+
+  const creative = getCreative(validAd);
+  const mediaFileUrl = getMediaFileUrl(creative);
+  const variationData = getVariationData(validAd);
+  const companionCreative = getCreativeCompanion(validAd);
+
+  videoPlayer.src = mediaFileUrl;
+  videoPlayer.play();
+
+  // listen the the en ended event to stop the ad and remove trackers
+  videoPlayer.addEventListener('ended', (e) => {
+    stopAd();
+  });
+
+  //once we have the parsed vast we can instanciate the vastTracker
+  vastTracker = new VASTTracker(null, validAd, creative);
+
+  //if we have a companion ad we instanciate the companionVastTracker
+  if (variationData.variation) {
+    companionVastTracker = new VASTTracker(
+      null,
+      validAd,
+      companionCreative,
+      variationData.variation
+    );
+  }
+
+  displayVariation(variationData);
+  initTrackers();
+};
+
+/**
+ * Stop the ad and remove all EventListeners
+ */
+
+export const stopAd = () => {
+  videoPlayer.src = mainContent;
+  videoPlayer.load();
+  videoPlayer.play();
+  variationContainer.classList.add('hidden');
```

Comment:
Perhaps it's not necessary to add a classlist for this
```suggestion
variationContainer.style.display = "none";
```